plugins/BEdita/Core/src/Model/Table/AsyncJobsTable.php

```diff
@@ -0,0 +1,270 @@
+ *
+ * @mixin \Cake\ORM\Behavior\TimestampBehavior
+ *
+ * @since 4.0.0
+ */
+class AsyncJobsTable extends Table
+{
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function defaultConnectionName()
+    {
+        if (in_array('async_jobs', ConnectionManager::configured())) {
+            return 'async_jobs';
+        }
+
+        return parent::defaultConnectionName();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @codeCoverageIgnore
+     */
+    public function initialize(array $config)
+    {
+        parent::initialize($config);
+
+        $this->setTable('async_jobs');
+
+        $this->setPrimaryKey('uuid');
+        $this->setDisplayField('payload');
+
+        $this->addBehavior('Timestamp', [
+            'events' => [
+                'Model.beforeSave' => [
+                    'created' => 'new',
+                    'modified' => 'always',
+                ],
+                'AsyncJob.complete' => [
+                    'completed' => 'always',
+                ],
+            ],
+        ]);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @codeCoverageIgnore
+     */
+    public function validationDefault(Validator $validator)
+    {
+        $validator
+            ->uuid('uuid')
+            ->allowEmpty('uuid', 'create');
+
+        $validator
+            ->requirePresence('service', 'create')
+            ->notEmpty('service');
+
+        $validator
+            ->naturalNumber('priority')
+            ->requirePresence('priority', 'create')
+            ->allowEmpty('priority');
+
+        $validator
+            ->allowEmpty('payload');
+
+        $validator
+            ->dateTime('not_before')
+            ->allowEmpty('not_before');
+
+        $validator
+            ->dateTime('not_after')
+            ->allowEmpty('not_after');
+
+        $validator
+            ->naturalNumber('max_attempts')
+            ->notEmpty('max_attempts');
+
+        $validator
+            ->dateTime('locked_until')
+            ->allowEmpty('locked_until');
+
+        $validator
+            ->dateTime('completed')
+            ->allowEmpty('completed');
+
+        return $validator;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @codeCoverageIgnore
+     */
+    protected function _initializeSchema(TableSchema $schema)
+    {
+        $schema->columnType('payload', 'json');
+
+        return parent::_initializeSchema($schema);
+    }
+
+    /**
+     * Lock an asynchronous job for execution.
+     *
+     * @param string $uuid UUID of job to be locked.
+     * @param mixed $duration Duration. By default, jobs are locked for 5 minutes.
+     * @return \BEdita\Core\Model\Entity\AsyncJob
+     */
+    public function lock($uuid, $duration = '+5 minutes')
+    {
+        return $this->getConnection()->transactional(function () use ($uuid, $duration) {
+            $entity = $this->get($uuid, ['finder' => 'pending']);
+            $entity->max_attempts -= 1;
```

Comment:
`'finder' => 'pending'` already adds the condition `WHERE ... AND max_attempts > 0`, so it shouldn't be necessary.